### PR TITLE
permissions: optimize API calls

### PR DESCRIPTION
**Pull request by Garfield-fr**
This optimization reduces the number of calls to the API when requesting permissions.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message title proposition: `permissions: optimize API calls`

**Comment by Garfield-fr** [comment]
> Commit message title proposition: `permissions: optimize API calls`

done

**Review by iGormilhit — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/routes/route-tool.service.ts [hunk 1]
```diff
@@ -211,6 +211,34 @@ export class RouteToolService {
...
   }
 
+  /**
+   * Check all permissions of the record
+   *
+   * @param record - Object: the resource object to check
+   * @param recordType - String: the record type
+   * @return Observable providing object with 2 attributes :
+   *     - 'can' - Boolean: to know if the resource could be deleted
+   *     - 'message' - String: the message to display if the record cannot be deleted
```
Comment (sebastiendeleze) [inline projects/admin/src/app/routes/route-tool.service.ts:221]: The explication can be more generalistic, not specific to deletion.

**Review by sebastiendeleze — APPROVED** [review]

File: projects/admin/src/app/routes/route-tool.service.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/admin/src/app/routes/route-tool.service.ts]: done

**Pull request merged**